duktape/src/main/jni/DuktapeContext.cpp

```diff
@@ -219,3 +228,52 @@ void DuktapeContext::bindInstance(JNIEnv *env, jstring name, jobject object, job
   // Pop the Duktape global object off the stack.
   duk_pop(m_context);
 }
+
+jobject DuktapeContext::call(JNIEnv *env, jstring name, jobject method, jobjectArray args) {
+  duk_push_global_object(m_context);
+  const JString instanceName(env, name);
+  if (!duk_get_prop_string(m_context, -1, instanceName)) {
+    duk_pop(m_context);
+    const jclass illegalArgumentException = env->FindClass("java/lang/IllegalArgumentException");
+    const std::string message = "A global object called " + instanceName.str() + " was not found";
+    env->ThrowNew(illegalArgumentException, message.c_str());
+    return nullptr;
+  }
+
+  // TODO: cache the call wrapper.
+
+  jclass methodClass = env->GetObjectClass(method);
+
+  const jmethodID getName = env->GetMethodID(methodClass, "getName", "()Ljava/lang/String;");
+  const JString methodName(env, static_cast<jstring>(env->CallObjectMethod(method, getName)));
+
+  // Push the function (property) onto the stack.
+  if (!duk_get_prop_string(m_context, -1, methodName)) {
+    // Pop the global object and our proxied instance.
+    duk_pop_2(m_context);
+    throwNoSuchMethod(env, "Method " + instanceName.str() + "." + methodName.str() + " not found");
```

Comment:
Maybe shove the word Duktape in here somewhere? Years from now Java programmers will look at their FooInterface and be like, "yes, the foo() method is totally found" not realizing that their interfaces are being implemented by JavaScript behind the scenes.